src/consumer.cpp

```diff
@@ -73,9 +73,14 @@ Consumer::~Consumer() {
         rebalance_error_callback_ = nullptr;
         close();
     }
-    catch (const Exception&) {
-        // If close throws just silently ignore until there's some
-        // logging facility (if any)
+    catch (const Exception& ex) {
+        auto& callback = get_configuration().get_log_callback();
```

Comment:
Can you make this `const auto&`? I prefer to explicitly indicate something is const instead of expecting the user to know that's the case.